app/models/audit/Auditable.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.audit
+
+import models.WithName
+import play.api.libs.json.{Json, Writes}
+
+trait Auditable { _: WithName =>
```

Comment:
Yeah, it's to ensure that it can only be mixed in with `WithName` because it sort of ensures that the `.toString` for the `code` has been overridden with an actual value then.